### PR TITLE
better docs for ActiveSupport::TestCase#assert_nothing_raised

### PR DESCRIPTION
**Pull request by wisetara**

### CODE REVIEW

File: activesupport/lib/active_support/test_case.rb [hunk 1]
```diff
@@ -66,10 +66,21 @@ def test_order
     alias :assert_not_respond_to :refute_respond_to
     alias :assert_not_same :refute_same
 
-    # Reveals the intention that the block should not raise any exception.
-    #
+
+    # Assertion that the block should not raise an exception.
+
```
Comment (rafaelfranca) [inline activesupport/lib/active_support/test_case.rb]: Missing `#` here.

File: activesupport/lib/active_support/test_case.rb [hunk 2]
```diff
@@ -66,10 +66,21 @@ def test_order
...
     #   assert_nothing_raised do
-    #     ...
+    #     perform_service(param: 'no_exception')
+    #   end
+
+    # With arguments, passes if evaluated code in the yielded block
+    # raises no specified exception.
+
+    #   assert_nothing_raised(Services::PerformService::MissingInputData) do
```
Comment (rafaelfranca) [inline activesupport/lib/active_support/test_case.rb]: We don't assert this. Actually we don't recommend passing arguments. We should just remove it.

Comment (wisetara) [inline activesupport/lib/active_support/test_case.rb]: Good to know! This IS why documentation is helpful. I'll revise.

**Comment by wisetara** [comment]
I haven't contributed here before, and I didn't squash my commits before I pushed again. Happy to do so, if you're interested in the PR.

**Comment by rafaelfranca** [comment]
:dizzy_face: seems some of our own tests are using the argument. I think we should keep it from now, deprecate it and change our own tests to not use it. Do you want to do this? If not it is fine, it is already a great first contribution :smile:! Thank you so much.

**Comment by wisetara** [comment]
I am so happy to make a first contribution! How about if I add back in the args and the documentation for using them--WITH a deprecation notice? How does that sound? I'd also be happy to see about sorting out the tests and removing the args in a separate PR.

**Comment by rafaelfranca** [comment]
:+1: to add it back, but it is better to not talk about the arguments in the documentation since we are going to deprecate and we don't want to people to use them. I agree that a separated is better.

**Comment by wisetara** [comment]
Okay, this time I squashed the commits, left out any mention of the ```(*args)``` that you'd like to deprecate, and I'll work on a PR for that deprecation. Thank you so much for the coaching and feedback!

**Comment by rafaelfranca** [comment]
Thanks!

**Pull request merged**